app/reducers/windows.ts

```diff
@@ -0,0 +1,212 @@
+/* eslint-disable */
+import { remote, shell, webContents } from 'electron';
+import { TYPES } from '@Actions/window_actions';
+import { TYPES as UI_TYPES } from '@Actions/ui_actions';
+import {
+    removeTrailingRedundancies
+} from '@Utils/urlHelpers';
+import { CONFIG } from '@Constants';
+import path from 'path';
+import logger from 'logger';
+import initialAppState from './initialAppState';
+
+const initialState = initialAppState.windows;
+
+/**
+ * Get the current window's webcontents Id. Defaults to `1` if none found.
+ * @return { Integer } WebContents Id of the curremt BrowserWindow webcontents.
+ */
+const getCurrentWindowId = () =>
+{
+    let currentWindowId = 1; // for testing
+    if ( remote )
+    {
+        currentWindowId = remote.getCurrentWindow().webContents.id;
```

Comment:
consider how this will work with tests (where there is not necessarily a window)